pkg/scheduler/framework/plugins/noderesources/balanced_allocation_test.go

```diff
@@ -183,9 +178,10 @@ func TestNodeResourcesBalancedAllocation(t *testing.T) {
 				st.MakePod().Node("node2").Labels(labels1).Obj(),
 			},
 			args:        config.NodeResourcesBalancedAllocationArgs{Resources: defaultResourceBalancedAllocationSet},
-			runPreScore: true,
+			runPreScore: false,
```

Comment:
Do we need `runPreScore: false` test cases with the best effort pod? Those are unrealistic (or uncommon) scenarios because we basically recommend people set the plugins at `multiPoint`, that is, it's a rare case that people disable plugin's PreScore only.

It'd prefer to remove such test cases (if other test cases cover things enough), or to change them to have pods with resource requests (if other test cases don't cover things enough).

Reply:
Other test cases cover enough ground, but @AxeZhan think we should keep these cases to verify the calculations in Score().

cc @macsko

Reply:
I meant "cover" equals whether other test cases verify the calculations in Score() enough.
If other test cases verify the calculations sufficiently, we can remove those, vs, if not, can we change the best effort pods into pods with request resources, which make the test cases realistic.

Reply:
Actually, it verifies the calculation sufficiently. I've changed one test to use pods with requested resources, while the remaining test case uses best-effort pods solely for calculating best-effort pods.

Reply:
>  it's a rare case that people disable plugin's PreScore only.

I think not.

As you mentioned, it's a rare case, and perhaps the user wants to disable PreScore to prevent skipping best-effort pods.

Reply:
> the remaining test case uses best-effort pods solely for calculating best-effort pods.

Then, can we remove those?

> shouldn't we change the Score func as well?
> As @sanposhiho mentioned, it's a rare case, and perhaps the user wants to disable PreScore to prevent skipping best-effort pods.

@macsko 's point makes sense to me, actually. We don't have a reason to implement the same at Score. Even though it's a rare scenario, we can make an effort to perform the best calculation, as long as the implementation isn't complicated by those.

Reply:
> Then, can we remove those?

Removed.

> @macsko 's point makes sense to me, actually. We don't have a reason to implement the same at Score. Even though it's a rare scenario, we can make an effort to perform the best calculation, as long as the implementation isn't complicated by those.

Sounds good, added.

cc @sanposhiho @macsko